### DISCUSSION TITLE
Title hyperlink in Citations

### DISCUSSION THREAD

**Discussion by bronyrafon** [microsoft/sample-app-aoai-chatGPT#1170] [General]
Hi,

I've deployed fresh today and I am using Azure AI Search with SharePoint and have included the metadata_spo_item_weburi identifier in my index. I thought I could just update AZURE_SEARCH_URL_COLUMN environment variable to be metadata_spo_item_weburi and then the title in the citation would then be clickable, but doesn't seem to work now?

I have an earlier deployment for something else, also using Azure AI Search and returning document URL's which I deployed on 14th March 2024 and haven't updated and the variable seems to work with this and the citation titles are clickable?

I notice in the readme it says AZURE_SEARCH_URL_COLUMN is not used currently? Is there anything I can do to make this work?

Gary.

**Comment by bronyrafon** [comment]
Ah, works if you populate - AZURE_SEARCH_FILENAME_COLUMN with metadata_spo_item_weburi

**Reply by jack-vinitsky** [reply]
I assume you are using the SharePoint Data Source in your Azure AI Search Service.  Wouldn't it make more sense to set AZURE_SEARCH_FILENAME_COLUMN to **metadata_spo_item_name** as that variable is supposed to represent the file name? 

The Title hyperlink in Citations is actually populated by index field that you map to AZURE_SEARCH_TITLE_COLUMN.

In our instances we use the built in Title column found in SharePoint Document Libraries.  In that scenario, we set AZURE_SEARCH_TITLE_COLUMN to Title. We then advise the business users to set the formal title of the document in that SharePoint column.  This allows them to hopefully put more meaningful information in that field rather then just the file name.  This helps for 2 reasons:

1. Creation of a Semantic Search Config when using Semantic or Semantic + Hybrid search.
2. We vector embed the Title column in addition to the document content via an extra embedding step in the Azure AI Search skillset.  Vector search is then done across two vector fields (vectorContent & vectorTitle).  Very helpful when you have documents with long meaningful titles like scientific publications.

**Reply by bronyrafon** [reply]
👍

**Reply by erolmam-75** [reply]
@bronyrafon @jack-vinitsky 
So how should I do this stepwize? As I have understood it, I do the following:

1. Define metadata for the documents in the Blob Storage: Key= **metadata_spo_item_name**, value=URL for the document e.-g. in sharepoint or other place
2. Change env setting AZURE_SEARCH_FILENAME_COLUMN to **metadata_spo_item_name**

Or should I use AZURE_SEARCH_URL_COLUMN?

I have ~700 files in my blob storage and trained AI Search index, but I can have these documents also in Sharepoint. I want to to have a way to link from the Citation page directly to the referred document for easier access to the user.

**Reply by jack-vinitsky** [reply]
> @bronyrafon @jack-vinitsky So how should I do this stepwize? As I have understood it, I do the following:
> 
> 1. Define metadata for the documents in the Blob Storage: Key= **metadata_spo_item_name**, value=URL for the document e.-g. in sharepoint or other place
> 2. Change env setting AZURE_SEARCH_FILENAME_COLUMN to **metadata_spo_item_name**
> 
> Or should I use AZURE_SEARCH_URL_COLUMN?
> 
> I have ~700 files in my blob storage and trained AI Search index, but I can have these documents also in Sharepoint. I want to to have a way to link from the Citation page directly to the referred document for easier access to the user.

You need to set 3 properties in the Environments file of the App Service:

AZURE_SEARCH_FILENAME_COLUMN ==> Field in your search index that contains the name of the file
AZURE_SEARCH_URL_COLUMN ==> Field in your search index that contains the URL to the file
AZURE_SEARCH_TITLE_COLUMN ==> Field in your search index that contains the formal title of the document.  If you don't have a separate field that holds the title, set it to the index field that has your file name.

**Comment by bronyrafon** [comment]
These are the  fields I have defined in my Index

  "fields":[
        { "name": "id", "type": "Edm.String", "key": true, "searchable": false },
        { "name": "metadata_spo_item_name", "type": "Edm.String", "key": false, "searchable": true, "filterable": false, "sortable": false, "facetable": false },
        { "name": "metadata_spo_item_weburi", "type": "Edm.String", "key": false, "searchable": false, "filterable": false,"sortable": false, "facetable": false },
        { "name": "metadata_spo_item_path", "type": "Edm.String", "key": false, "searchable": false, "filterable": false, "sortable": false, "facetable": false },
        { "name": "metadata_spo_item_content_type", "type": "Edm.String", "key": false, "searchable": false, "filterable": true, "sortable": false, "facetable": true },
        { "name": "metadata_spo_item_last_modified", "type": "Edm.DateTimeOffset", "key": false, "searchable": false, "filterable": false, "sortable": true, "facetable": false },
        { "name": "metadata_spo_item_size", "type": "Edm.Int64", "key": false, "searchable": false, "filterable": false, "sortable": false, "facetable": false },
        { "name": "content", "type": "Edm.String", "searchable": true, "filterable": false, "sortable": false, "facetable": false }

This is where I ended up with the environment variables:

AZURE_SEARCH_TITLE_COLUMN - metadata_spo_item_name
AZURE_SEARCH_URL_COLUMN - metadata_spo_item_weburi
AZURE_SEARCH_CONTENT_COLUMNS - content
AZURE_SEARCH_FILENAME_COLUMN - id

I tweaked the chat.tsx file so that the citation title which is clickable now appears as blue underlined text so that users know to click on it as a link..

**Reply by erolmam-75** [reply]
Thanks a lot. both of you 🙏
I managed to re-create this for one document and today I work to do this for all doc's. I only use Azure portal, so no tweaking from my side 😊😊

**Reply by jack-vinitsky** [reply]
@bronyrafon I like your change to the chat.tsx file to make it blue and underlined.

I would set AZURE_SEARCH_FILENAME_COLUMN to metadata_spo_item_name so that the file name appears in the references underneath the response.

![image](https://github.com/user-attachments/assets/d22d38d8-dc63-4e19-9c37-cec3a530acb3)
